---

```{r global_setup, echo = FALSE, cache = FALSE, warning = FALSE}

    opts_chunk$set(cache = FALSE, fig.path = 'figures/', autodep = TRUE, fig.cap = "", fig.width = 8, fig.height = 8, dev = c("cairo_pdf"), echo = FALSE, warning = FALSE, message = FALSE)
    dep_auto()

```

```{r setup}

    for (package in c("knitr", "markdown", "edgeR", "stringr")){
        library(package, character.only = TRUE)
    }

    source("src/main.r")

    # There are two data objects, filone represents the data set used in Filone 2013, yang_a represents the data set used in Yang 2010 (Yang, Z., Bruno, D. P., Martens, C. A., Porcella, S. F., & yang, B. (2010). Simultaneous high-resolution analysis of vaccinia virus and host cell transcriptomes by deep RNA sequencing. Proceedings of the National Academy of Sciences of the United States of America, 107(25), 11513–8. doi:10.1073/pnas.1006594107)
    filone <- get_data_set("data/filone_counts.txt")
    filone <- add_ma(filone)

    yang_a <- get_data_set("data/yang_a_counts.txt")
    yang_a <- add_ma(yang_a)

    # Genes with a fold change greater than 2 and HSF1 targets as identified by ChIP-Seq in Mendillo 2012 (Mendillo, M. L., Santagata, S., Koeva, M., Bell, G. W., Hu, R., Tamimi, R. M., Fraenkel, E., et al. (2012). HSF1 drives a transcriptional program distinct from heat shock to support highly malignant human cancers. Cell, 150(3), 549–62. doi:10.1016/j.cell.2012.06.031)
    mendillo_hsf1_targets <- scan("data/mendillo_hsf1_targets.txt", "character")

    red <- "#CA0020"

    cutoff <- list(counts = 20, fold_change = 1.5)
    is_valid_gene <- list(filone = filone$normalized_counts$hpi_6 > cutoff$counts & filone$hpi_6$ma$m > cutoff$fold_change,
                          yang_a = yang_a$normalized_counts$hpi_4 > cutoff$counts & yang_a$hpi_4$ma$m > cutoff$fold_change)
    valid_gene_names <- list(filone = rownames(filone$normalized_counts[is_valid_gene$filone,]),
                             yang_a = rownames(yang_a$normalized_counts[is_valid_gene$yang_a,]))

    upregulated_genes <- list()
    upregulated_genes$filone <- filone$normalized_counts[valid_gene_names$filone,]
    upregulated_genes$filone$log2_fold_change_6hpi <- filone$hpi_6$ma[valid_gene_names$filone, "m"]
    upregulated_genes$filone$log10_abundance_6hpi <- filone$hpi_6$ma[valid_gene_names$filone, "a"]
    upregulated_genes$filone$hsf1_target <- ifelse(valid_gene_names$filone %in% mendillo_hsf1_targets, "yes", "unknown")
    # write.table(upregulated_genes$filone, file = "data/upregulated_genes_filone.txt", sep = "\t", quote=FALSE, row.names=TRUE, col.names=TRUE)

    upregulated_genes$yang_a <- yang_a$normalized_counts[valid_gene_names$yang_a,]
    upregulated_genes$yang_a$log2_fold_change_4hpi <- yang_a$hpi_4$ma[valid_gene_names$yang_a, "m"]
    upregulated_genes$yang_a$log10_abundance_4hpi <- yang_a$hpi_4$ma[valid_gene_names$yang_a, "a"]
    upregulated_genes$yang_a$hsf1_target <- ifelse(valid_gene_names$yang_a %in% mendillo_hsf1_targets, "yes", "unknown")
    # write.table(upregulated_genes$yang_a, file = "data/upregulated_genes_yang_a.txt", sep = "\t", quote=FALSE, row.names=TRUE, col.names=TRUE)

    length(rownames(upregulated_genes$filone))
    length(rownames(upregulated_genes$yang_a))
    length(intersect(rownames(upregulated_genes$filone), rownames(upregulated_genes$yang_a)))
```

```{r plots, fig.width=11, fig.height=11}

    plot(filone$hpi_6$ma[,c("a","m")], pch = 16, col = ifelse(is_valid_gene$filone, "black", "grey70"), xlab="log10 normalized counts", ylab="log2 fold change (at 6hpi)", family = "Arial", xlim = c(0,6), ylim = c(-3,8), xpd = FALSE)
    points(filone$hpi_6$ma[is_valid_gene$filone,][mendillo_hsf1_targets, "a"], filone$hpi_6$ma[is_valid_gene$filone,][mendillo_hsf1_targets, "m"], col = red, pch = 16)
    annotate_points(filone$hpi_6$ma[is_valid_gene$filone,][mendillo_hsf1_targets, "a"], filone$hpi_6$ma[is_valid_gene$filone,][mendillo_hsf1_targets, "m"], mendillo_hsf1_targets, col= red, pos = 4)
    legend("topright", c("Upregulated", "HSF1 target") , pch = 16, col = c("black", red), cex = 1.5)

    plot(yang_a$hpi_4$ma[,c("a","m")], pch = 16, col = ifelse(is_valid_gene$yang_a, "black", "grey70"), xlab="log10 normalized counts", ylab="log2 fold change (at 6hpi)", family = "Arial", xlim = c(0,6), ylim = c(-3,8), xpd = FALSE)
    points(yang_a$hpi_4$ma[is_valid_gene$yang_a,][mendillo_hsf1_targets, "a"], yang_a$hpi_4$ma[is_valid_gene$yang_a,][mendillo_hsf1_targets, "m"], col = red, pch = 16)
    annotate_points(yang_a$hpi_4$ma[is_valid_gene$yang_a,][mendillo_hsf1_targets, "a"], yang_a$hpi_4$ma[is_valid_gene$yang_a,][mendillo_hsf1_targets, "m"], mendillo_hsf1_targets, col= red, pos = 4)
    legend("topright", c("Upregulated", "HSF1 target") , pch = 16, col = c("black", red), cex = 1.5)

    df_filone <- bring_to_front(filone$log2_fold_changes[valid_gene_names$filone, ], mendillo_hsf1_targets[mendillo_hsf1_targets %in% valid_gene_names$filone])
    plot_lines(df_filone[, -5], col = ifelse(rownames(df_filone) %in% mendillo_hsf1_targets, transparent(red), transparent("black")), ylab = "log2 fold change (at 6hpi)", xlab="hour post-infection", col_names=str_to_num(colnames(filone$log2_fold_changes[, -5])), at_x_axis = str_to_num(colnames(filone$log2_fold_changes[, -5])), family = "Arial", ylim = c(-3,8))
    text(6,filone$log2_fold_changes[valid_gene_names$filone, ][mendillo_hsf1_targets, 4], labels = mendillo_hsf1_targets, col = red, pos = 4, font = 2, xpd = NA, cex = 1.2)

    df_yang_a <- bring_to_front(yang_a$log2_fold_changes[valid_gene_names$yang_a, ], mendillo_hsf1_targets[mendillo_hsf1_targets %in% valid_gene_names$yang_a])
    plot_lines(df_yang_a, col = ifelse(rownames(df_yang_a) %in% mendillo_hsf1_targets, transparent(red), transparent("black")), ylab = "log2 fold change (at 6hpi)", xlab="hour post-infection", col_names=str_to_num(colnames(yang_a$log2_fold_changes)), at_x_axis = str_to_num(colnames(yang_a$log2_fold_changes)), family = "Arial", ylim = c(-3,8))
    text(4,yang_a$log2_fold_changes[valid_gene_names$yang_a, ][mendillo_hsf1_targets, 5], labels = mendillo_hsf1_targets, col = red, pos = 4, font = 2, xpd = NA, cex = 1.2)

```

```{r embed_fonts}

    # Embed fonts to pdf due to some issue in Cairo
    library(extrafont)
    sapply(dir(opts_chunk$get("fig.path"), pattern="*.pdf", full.name = TRUE), embed_fonts)

```